volumina/widgets/layerDialog.py

```diff
@@ -27,146 +27,82 @@
 # PyQt
 from PyQt5 import uic
 from PyQt5.QtWidgets import QDialog
+from PyQt5.QtCore import Qt
 from functools import partial
+from typing import Callable
+from pathlib import Path
+
+from volumina.widgets.thresholdingWidget import ThresholdingWidget
 
 import logging
 
 logger = logging.getLogger(__name__)
 
-# ===----------------------------------------------------------------------------------------------------------------===
-
 
-class GrayscaleLayerDialog(QDialog):
-    def __init__(self, layer, parent=None):
-        QDialog.__init__(self, parent)
-        p = os.path.split(os.path.abspath(__file__))[0]
-        uic.loadUi(p + "/ui/grayLayerDialog.ui", self)
+class LayerDialog(QDialog):
+    def __init__(self, ui_file_name: str, layer, parent=None):
+        super().__init__(parent)
+        base_path = Path(__file__).resolve().parent
+        ui_path = base_path.joinpath("ui").joinpath(ui_file_name)
+        uic.loadUi(ui_path.as_posix(), self)
         self.setLayername(layer.name)
+        self.layer = layer
 
-        def dbgPrint(a, b):
-            layer.set_normalize(0, (a, b))
-            logger.debug("normalization changed to [%d, %d]" % (a, b))
+    def initialize_range_widgets(self, datasourceIdx: int, thresholding_widget: ThresholdingWidget, autorange_checkbox):
+        def handleRangeChanged(a, b):
+            self.layer.set_normalize(datasourceIdx, (a, b))
+            print(f"normalization changed for channel={datasourceIdx} to [{a}, {b}]")
```

Comment:
hang on, I would prefer to either have this a debug log message, or gone completely